README.MD
See its description at AlliedModders' Forum
=============================

Addons zz's Multi-Mod Server
https://forums.alliedmods.net/showthread.php?t=273018


ChangeLog.txt
*Change Log
- All notable changes to this project will be documented in this file.
- This project adheres to [Semantic Versioning](http://semver.org/).

2016-02-24 - v4.0.2
	* Minor misspellings.

2015-08-30 - v4.0.1
	* Added new commandmenu options.
	* Fixed messages menu sounds and messages.

2015-08-27 - v4.0
	* Separated the Amx_Ultra plugin from the Addons Multi-Mod.

2015-08-22 - v3.0
	* Currently it is tested under Counter-Strike and Counter-Strike Condition Zero
	* Added support to control PodBot and CSBot throw commandmenu.txt
	* Log chat messages of: say, say_team, amx_say, ...

2015-08-17 - v2.0
    * Added Galileo 1.1.290 that is a feature rich map voting plugin.
    * In the last five minutes or be asked before a vote by the command "say votemod", 
	   creates a vote to select what will be the Mod played in the next changelevel/restart.
    * Developed a multi-page votemod menu system to display until 100 mods.
    * Added a currentmod.ini file to save current active mod id and load it at server start.
    * Changes the mapcycle, if and only if a custom mod mapcycle was created.
    * Made the votemod keep the current mod if less than 30% of players voted.
    * Made "Extend current map" right after choose, not to restart the game at the current map.
    * Made the "currentmod.ini" store mod ids related to the mods order at "multimod.ini".
    * Fixed current mod message is displaying "Next Mod: ".
    * Made "Next Mod: " message display there is no actived mod, when there is not.
    * When the min vote time is not reached/disabled, display e message informing that.

2015-08-12 v1.5
	* Added Dragon Ball Mod v1.3
	* New multi-mod_core with improved server control.
	* Fixed daily_maps incompatibility with nextmap.
	* Placed multi-mod_plugin and info to its originals plugins nextmap and cmdmenus.

2015-08-10 1.4
	* Added Gun-Game Mod v2.13c
	* Added CSDM (Death-Match) v2.1.3c
	* Restaured the broken restart menu.

2015-08-10 v1.3
	* Added pain_shock_free plugins that disables the floor wander when taking shots.
	* Added support for Superheros Mod 1.2.1
	* Added support for Predator Mod_B2 2.1
	* Added support for Ultimate Warcraft Mod 3
	* Added support for Knife Arena Mod 1.2
	* Added the mod Zombie Mod 5.08a with new game modes:
	* Nemesis:
	   The first zombie may turn into a Nemesis, a powerful fast-moving
	   beast. His goal is to kill every human while sustaining the gunfire.
	* Survivor:
	   Everyone became a zombie except him. The survivor gets a machinegun
	   with unlimited ammo and has to stop the never-ending army of undead.
	* Multiple Infection:
	   The round starts with many humans infected, so the remaining players
	   will have to act quickly in order to control the situation.
	* Swarm Mode:
	   Half of the players turn into zombies, the rest become immune and
	   cannot be infected. It's a battle to death.
	* Plague Mode: [bonus]
	   A full armed Survivor and his soldiers are to face Nemesis and
	   his zombie minions. The future of the world is in their hands.

2015-07-27 v1.2
	* Added new Command Menu (h button on the game) with support:
	- Enabling Superheros Mod, Mod Predator, Knife Arena Mod and Ultimate Warcraft Mod.
	- Greater control of the server and finish the round in a tie, or the CT's or TR's winning.
	- Binds configured as walk-continue and fast change.
	- Enabling PODBot and commands Superheros Mod.
	- Access to the top 15, remaining time of the map and the current map.
	- Control of PODBots settings such as quota, time, difficulty, kill, remove, weapons mode and etc ...
	- Gravity Change, friendly fire, equilibrium times, times limit ...
	- Added support for linux and windows servers.
	- Added support for Mac OS PODBot and Linux.
	* Added hlds.bat file and hlds.sh to create served by command line in Windows on Linux.
	* Added all the code sources used:
	- A total of 397 plugins to sources.
	- player_wanted that pays rewards for CT's and TR's most sought after.
	- amx_plant_bonus giving a bonus in cash, who plant the C4.
	- Amx Mod X 1.82 and PODBot V3.0 metamod Build 22.
	- usurf that provides help and other things to surf maps.
	- cssurfboards adding a surfboard, (amx_createnpc).
	- lastmanbets Plugin bets when left over 1x1.
	- BombSite_Radar to see where the locals to plant the bomb
	- bad_camper that punishes who does camper indiscriminately.
	- multi-mod_core, amx_exec, head_shot_announcer, grentrail, parachute, knife_duel, 
	  amx_chicken, adv_killstreak, countdown_exec, ... 

2015-07-09
	* Hlds.bat: Created a file that creates a server throw command line. That is
     useful because a command line server usually is 50% more efficient in the use
     procesador.

website/default.lang.en.php
<?php
/* 
------------------
Language: ENGLISH
------------------
*/

$lang = array();

$lang[ 'HEADER' ] = <<<EOD
    <meta name="description"content="The Addons Multi-Mod is an add-in for Counter-Strike and Counter-Strike Condition Zero games. The Addons Multi-Mod contains numerous mods and comes standard configured for maximum performance and server control.">
    <title>Addons Multi-Mod - For Counter-Strike - CS </title>
    <meta itemprop="name" content="Addons Multi-Mod - For Counter-Strike - CS ">
    <meta itemprop="description"content="The Addons Multi-Mod is an add-in for Counter-Strike and Counter-Strike Condition Zero games. The Addons Multi-Mod contains numerous mods and comes standard configured for maximum performance and server control.">
EOD;

$lang[ 'MENU_PRINCIPAL' ] = <<<EOD
<div id="barraDoIdioma" style="text-align:left; margin: 0 auto; ">Language/Idioma: 
    <a href="default.php?lang=en">English</a>
    - - 
    <a href="default.php?lang=pt">Português</a>
</div>
<div id="conteinerDoLogo" align="center" style="display: table; min-height:150px; max-width: 1360px; width:98%; text-align: center; margin: 0 auto; align-content:center; ">
    <a href="http://www.addons.zz.mu/">
    <span id="logoPrincipal"></span></a>
</div>
EOD;

if( $_SESSION[ 'screen_width' ] < 800 )
{
    $lang[ 'MENU_PRINCIPAL' ] = $lang[ 'MENU_PRINCIPAL' ] . <<<EOD
<div class="tableContainer " style="border:double; margin: 0 auto; border-color:#232323; ">
    <div id="tableRow">
        <div class="tableColumn " style="border-right:double; border-bottom:double; padding:5px; border-color:#232323; ">
            <a href="http://www.addons.zz.mu/"><b>Home Page</b></a>
        </div>
        <div class="tableColumn " style="border-bottom:double; padding:5px; border-color:#232323; ">
            <a href="https://github.com/Addonszz/AddonsMultiMod/issues"><b>Forum - Questions - Request Features</b></a>
        </div>
    </div>
    <div id="tableRow">
        <div class="tableColumn " style="border-right:double; border-color:#232323; padding:5px; ">
            <a href="#installation"><b>Installation</b></a>
        </div>
        <div class="tableColumn " style="border-right:double; border-color:#232323; padding:5px; ">
            <a href="https://github.com/Addonszz/AddonsMultiMod/releases"><b>Downloads</b></a>
        </div>
        <div class="tableColumn " align="center" style="padding:5px; border-color:#232323; ">
            <a href="https://github.com/Addonszz"><b>Contact</b></a>
        </div>
    </div>
</div>
EOD;
} else
{
    $lang[ 'MENU_PRINCIPAL' ] = $lang[ 'MENU_PRINCIPAL' ] . <<<EOD
<div class="tableContainer " style="border:double; margin: 0 auto; border-color:#232323;">
    <div id="tableRow">
        <div class="tableColumn " style="border-right:double; padding:5px; border-color:#232323;">
            <a href="http://www.addons.zz.mu/"><b>Home Page</b></a>
        </div>
        <div class="tableColumn " style="border-right:double; padding:5px; border-color:#232323;">
            <a href="https://github.com/Addonszz/AddonsMultiMod/issues"><b>Forum - Questions - Request Features</b></a>
        </div>
        <div class="tableColumn " style="border-right:double; padding:5px; border-color:#232323;">
            <a href="#installation"><b>Installation</b></a>
        </div>
        <div class="tableColumn " style="border-right:double; padding:5px; border-color:#232323;">
            <a href="https://github.com/Addonszz/AddonsMultiMod/releases"><b>Downloads</b></a>
        </div>
        <div class="tableColumn " style="padding:5px; border-color:#232323;">
            <a href="https://github.com/Addonszz"><b>Contact</b></a>
        </div>
    </div>
</div>
EOD;
}

$lang[ 'DESCRICAO_INICIAL' ] = <<<EOD
The Addons Multi-Mod is an add-in for &quot;Counter-Strike 1.6&quot; and &quot;Counter-Strike: Condition Zero&quot; game.
<p>
The Addons Multi-Mod contains numerous mods and comes standard configured for maximum performance and server control.</p>
EOD;

$lang[ 'PRIMEIRA_COLUNA' ] = <<<EOD
<p><span style="font-size:40px">About Addons Multi-Mod? See here:</span>
<p><a href="https://github.com/Addonszz/AddonsMultiMod">https://github.com/Addonszz/AddonsMultiMod</a></p>
</p>
<p><span style="font-size:40px">About Amx Ultra? See here:</span>
<p><a href="https://github.com/Addonszz/Amx_Ultra">https://github.com/Addonszz/Amx_Ultra</a></p>
</p>
<p>Basic information on the latest version of <strong><font color="red"><u>Addons Multi-Mod v2.0</u></font></strong>
    released on 12/08/2015: </p>
* Galileo 1.1.290 that is a feature rich map voting plugin.<br>
* In the last five minutes or be asked before a vote by the command "say votemod", creates a vote to select what will be the Mod played in próprimo round.<br>
*  Developed a multi-page votemod menu system to display until 100 mods.<br>
*  Added a currentmod.ini file to save current active mod id and load it at server start.
<p><img src="recursos/2015-08-16_14-08_Counter-Strike.jpg" width="600" ></p>
<p><img src="recursos/2015-08-16_14-08_Counter-Strike(2).jpg" width="600" ></p>
*  Changes the mapcycle, if and only if a custom mod mapcycle was created.<br>
*  Made the votemod keep the current mod if less than 30% of players voted.<br>
*  Made "Extend current map" right after choose, not to restart the game at the current map.<br>
*  Made the "currentmod.ini" store mod ids related to the mods order at "multimod.ini".<br>
*  Fixed current mod message is displaying "Next Mod: ".<br>
*  Made "Next Mod: " message display there is no actived mod, when there is not.<br>
*  When the min vote time is not reached/disabled, display e message informing that.

<p>Version of <strong><font color="red"><u>Addons Multi-Mod v1.5</u></font></strong>
    released on 12/08/2015: <br>

* Added Dragon Ball Mod v1.3<br>
* New multi-mod_core with improved server control.<br>
* Fixed daily_maps incompatibility with nextmap.<br>
* Placed multi-mod_plugin and info to its originals plugins nextmap and cmdmenus.

<p>Version of <strong><font color="red"><u>Addons Multi-Mod v1.4</u></font></strong>
    released on 10/08/2015: <br>
* Added pain_shock_free plugins that disables the slow walking when taking shots.<br>
* Added new Command Menu (h button on the game) with support:<br>
* Enabling Superheros Mod, Mod Predator, Knife Arena Mod and Ultimate Warcraft Mod.<br>
* Greater control of the server and finish the round in a tie, or the CT's or TR's winning.

<p><img src="recursos/2015-07-28_05-18_Untitled.jpg" width="362" height="424" alt=""/></p>
* Binds configured as walk-continue and fast change.<br>
* Enabling PODBot and commands Superheros Mod.<br>
* Access to the top 15, remaining time of the map and the current map.<br>
* Control of PODBots settings such as quota, time, difficulty, kill, remove, weapons mode and etc ...<br>
* Gravity Change, friendly fire, equilibrium times, times limit ...</p>

<p><img src="instalacao/2015-07-27_04-02_Counter-Strike.jpg" width="360" height="334"></p>

* Added support for linux and windows servers.<br>
* Added support for Mac OS PODBot, Linux and Windows.<br>
* Added support for Zombie Plague Mod 5.08a<br>
* Added support for Superheros Mod 1.2.1<br>
* Added support for CSDM (Death-Match) v2.1.3c<br>
* Added support for Gun-Game Mod v2.13c<br>
* Added support for Predator Mod_B2 2.1<br>
* Added support for Ultimate Warcraft Mod 3<br>
* Added support for Knife Arena Mod 1.2<br>
* Added hlds.bat file and hlds.sh to create served by command line in Windows on Linux.

<p><img src="recursos/2015-07-28_04-18_Half-Life.jpg" width="465" height="118" alt=""/></p>

*** Added all the code sources used:<br>
* A total of 450 plugins to sources.<br>
* player_wanted that pays rewards for CT's and TR's most sought after.<br>
* amx_plant_bonus giving a bonus in cash, who plant the C4.<br>
* Amx Mod X 1.82 and PODBot V3.0 metamod Build 22.<br>
* usurf that provides help and other things to surf maps.<br>
* cssurfboards adding a surfboard, (amx_createnpc).

<p><img src="recursos/2015-07-28_03-39_Counter-Strike.jpg" width="607" height="711" alt=""/></p>

* lastmanbets Plugin bets when left over 1x1.<br>
* BombSite_Radar to see where the locals to plant the bomb<br>
* bad_camper that punishes who does camper indiscriminately.<br>
* multi-mod_core, amx_exec, head_shot_announcer, grentrail, parachute, knife_duel, amx_chicken,
adv_killstreak, countdown_exec, ...<br>
* Total possible modification of available resources addons.

<p><a href="https://github.com/Addonszz/AddonsMultiMod/blob/master/gamemod_common/addons/amxmodx/configs/plugins.ini">See more info here.</a></p>
EOD;

$lang[ 'SEGUNDA_COLUNA' ] = <<<EOF
<p>It works in <a name="installation"><strong><font color="red"><u>Counter-Strike e Counter-Strike Condition Zero</u></font></strong></a>  updated.</p>

<p><span style="font-size:66px">To install it</span><br>Download the binaries 
<a href="https://github.com/Addonszz/Amx_Ultra/releases/download/v1.0/amx_ultra_plugin.zip">amx_ultra_plugin.zip</a>, 
<a href="https://github.com/Addonszz/Amx_Ultra/releases/download/v1.0/amx_ultra_resources.zip">amx_ultra_resources.zip</a>, 
<a href="https://github.com/Addonszz/AddonsMultiMod/releases/download/v4.0/addons_resources.zip">addons_resources.zip</a>, 
<a href="https://github.com/Addonszz/Amx_Ultra/archive/master.zip">Amx_Ultra-master.zip</a> 
and 
<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>, 
then just unzip and put the content of <br>
"<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>/AddonsMultiMod-master/cstrike" or <br>
"<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>/AddonsMultiMod-master/czero" and <br>
"<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>/AddonsMultiMod-master/gamemod_common", <br>
"<a href="https://github.com/Addonszz/Amx_Ultra/archive/master.zip">Amx_Ultra-master.zip</a>/Amx_Ultra-master", <br>
"<a href="https://github.com/Addonszz/Amx_Ultra/releases/download/v1.0/amx_ultra_plugin.zip">amx_ultra_plugin.zip</a>", <br>
"<a href="https://github.com/Addonszz/AddonsMultiMod/releases/download/v4.0/addons_resources.zip">addons_resources.zip</a>" <br>
and 
"<a href="https://github.com/Addonszz/Amx_Ultra/releases/download/v1.0/amx_ultra_resources.zip">amx_ultra_resources.zip</a>" <br>
at your Counter-Strike's cstrike or czero folder, 
replacing existents files. The Counter-Strike's cstrike or czero folder usually is at:</p>

<p><u>C:\Program Files (x86)\Steam\SteamApps\common\Half-Life\cstrike</u></p>
<p><u>C:\Program Files (x86)\Steam\SteamApps\common\Half-Life\czero</u></p>

If your game is different than "cstrike" or "czero", edit your "liblist.gam" at your 
gamemod folder as the "liblist.gam" at "cstrike" or "czero" folder inside <br>
"<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>/AddonsMultiMod-master".

<p>After installing addons, just set your STEAM ID user.</p>

<p>Please update the file:</p>

<p>***users (user.ini) at folder gamemod/addons/amxmodx/configs</p>

<p>After installing addons, just set your RCON password.</p>

<p>***autoexec (autoexec.cfg) at folder gamemod. </p>

<p>Following the instructions contained therein.</p>

<p>Note 1: In order to use the commandmenu (key h in game). Every server 
administrator should have the file <br>
"<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>/gamemod_common" <br> 
above installed, and with the RCON password configured at his own Game Mod copy. <br><br>
An admin can too just have the folder <br>
"<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>/gamemod_common/admin" <br>
with its contents and the files <br>
"<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>/gamemod_common/commandmenu.txt" <br>
and 
"<a href="https://github.com/Addonszz/AddonsMultiMod/archive/master.zip">AddonsMultiMod-master.zip</a>/gamemod_common/autoexec.cfg", <br>
with the RCON password configured.</p>

<p>Note 2: The PASSWORD at podbotconfig.cfg at your gamemod folder, serves 
to create waypoint using the linstenserver (play offline at new game) and add 
podbots. But who has rcon authentication can also control the podbots.</p>

<p>If you have trouble configuring the server, this is an awesome tutorial on setting an updated Steam server that works with any type of customer:</p>

<p>
    <a href="http://steamcommunity.com/sharedfiles/filedetails/?id=340974032">http://steamcommunity.com/sharedfiles/filedetails/?id=340974032</a>
</p>

<p>
    <a href="https://developer.valvesoftware.com/wiki/SteamCMD">https://developer.valvesoftware.com/wiki/SteamCMD</a>
</p>

<p><span style="font-size:66px">See here</span> newer instructions for installation:
<a href="https://github.com/addonszz/AddonsMultiMod#to-install-it">https://github.com/addonszz/AddonsMultiMod#to-install-it</a>
</p>

EOF;
?>